collectors/plugins.d/pluginsd_parser.c

```diff
@@ -51,13 +51,15 @@ PARSER_RC pluginsd_chart_action(void *user, char *type, char *id, char *name, ch
     RRDHOST *host = ((PARSER_USER_OBJECT *) user)->host;
     uuid_t  *guid = &((PARSER_USER_OBJECT *)user)->guid;
 
+    int is_guid_null = uuid_is_null(*guid);
     st = rrdset_create_custom(
         host, type, id, name, family, context, title, units, plugin, module, priority, update_every, chart_type,
         host->rrd_memory_mode, host->rrd_history_entries, 0,
-        uuid_is_null(*guid) ? NULL : guid);
+        is_guid_null ? NULL : guid);
 
 #ifdef ENABLE_DBENGINE
-    uuid_clear(((PARSER_USER_OBJECT *) user)->guid);
+    if (likely(!is_guid_null))
```

Comment:
In the case of collectors via plugins.D (and not streaming) the `likely` is going to hit the misprediction penalty 100% of the time. I think we should remove any hints in this check and let the CPU train itself depending on whether it is log replay, streaming or collectors.